File: SpringBootBackend/src/main/java/me/devziyad/springbootbackend/booking/Booking.java
package me.devziyad.springbootbackend.booking;

import jakarta.persistence.*;
import lombok.*;
import me.devziyad.springbootbackend.common.BookingStatus;
import me.devziyad.springbootbackend.ride.Ride;
import me.devziyad.springbootbackend.user.User;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "ride_id")
    private Ride ride;

    @ManyToOne(optional = false)
    @JoinColumn(name = "rider_id")
    private User rider;

    @Column(nullable = false)
    private Integer seatsBooked;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private BookingStatus status = BookingStatus.PENDING;

    @Column(nullable = false)
    private BigDecimal costForThisRider;

    @Column(nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column
    private LocalDateTime cancelledAt;
}

File: flutter_application_1/lib/models/booking.dart
class Booking {
  final int id;
  final int rideId;
  final int riderId;
  final String riderName;
  final int seatsBooked;
  final String status;
  final double costForThisRider;
  final DateTime createdAt;
  final DateTime? cancelledAt;

  Booking({
    required this.id,
    required this.rideId,
    required this.riderId,
    required this.riderName,
    required this.seatsBooked,
    required this.status,
    required this.costForThisRider,
    required this.createdAt,
    this.cancelledAt,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['id'] as int,
      rideId: json['rideId'] as int,
      riderId: json['riderId'] as int,
      riderName: json['riderName'] as String,
      seatsBooked: json['seatsBooked'] as int,
      status: json['status'] as String,
      costForThisRider: (json['costForThisRider'] as num).toDouble(),
      createdAt: DateTime.parse(json['createdAt'] as String),
      cancelledAt: json['cancelledAt'] != null
          ? DateTime.parse(json['cancelledAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'rideId': rideId,
      'riderId': riderId,
      'riderName': riderName,
      'seatsBooked': seatsBooked,
      'status': status,
      'costForThisRider': costForThisRider,
      'createdAt': createdAt.toIso8601String(),
      'cancelledAt': cancelledAt?.toIso8601String(),
    };
  }
}

